### PR TITLE
fix: not inject `metadata.namespace` in manifests rendered with kustomize

### PR DESCRIPTION
**Pull request by renzodavid9**
Fixes: #8403

**Description**
This is to to fix a v1 regression: manifests rendered with kustomize shouldn't have a `metadata.namespace` field injected by skaffold.

A new `InjectNamespace` property is added to decide if Skaffold should inject a namespace by itself or not. For [8318](https://github.com/GoogleContainerTools/skaffold/issues/8318) we can use this new property, reading its value from a flag or configuration.

**Follow-up Work**
[8318](https://github.com/GoogleContainerTools/skaffold/issues/8318)

### CODE REVIEW

File: pkg/skaffold/render/renderer/util/util.go [hunk 1]
```diff
@@ -45,7 +45,9 @@ type GenerateHydratedManifestsOptions struct {
 func GenerateHydratedManifests(ctx context.Context, out io.Writer, builds []graph.Artifact, g generate.Generator, labels map[string]string, ns string, opts GenerateHydratedManifestsOptions) (manifest.ManifestList, error) {
 	// Generate manifests.
 	rCtx, endTrace := instrumentation.StartTrace(ctx, "Render_generateManifest")
-	manifests, err := g.Generate(rCtx, out)
+	// TODO(https://github.com/GoogleContainerTools/skaffold/issues/8318): revisit this code to better tackle this case for kustomize.
+	// We could have a different renderer for kustomize that doesn't inject namespaces.
+	kustomizeManifests, otherManifests, err := g.Generate(rCtx, out)
```
Comment (aaron-prindle) [inline pkg/skaffold/render/renderer/util/util.go:50]: NOTE: @ericzzzzzzz has added a seperate `kustomize` renderer in his PR here:
https://github.com/GoogleContainerTools/skaffold/pull/8365

Comment (renzodavid9) [reply pkg/skaffold/render/renderer/util/util.go:50]: Yes! 😄 with it we can tackle this case better.

**Comment by aaron-prindle** [comment]
@renzodavid9 Seems this needs a rebase now after merging #8365

**Comment by renzodavid9** [comment]
> @renzodavid9 Seems this needs a rebase now after merging #8365

Done! I rebased with the latest changes and changed the code to tackle this. Let me know what do you think.

**Review by ericzzzzzzz — APPROVED** [review]

**Review by aaron-prindle — APPROVED** [review]
LGTM!

**Pull request merged**